### PR TITLE
Align LpPool ceil-mode docs and tests with AveragePool, MaxPool

### PR DESCRIPTION
**Pull request by jonahbernard**
# Description

- Clarify the LpPool `ceil_mode` docstring so it matches the expected behavior in Pytorch and runtimes (ORT) 
- Refresh generated operator docs to pick up the new wording  
- Add a shape-inference unit test that covers LpPool with padding, stride, and `ceil_mode=1`  
- Add a backend node test mirroring the ORT runtime guard when the last window would land entirely in padding  

# Motivation and Context

- @liqunfu's guard (ORT PR matching ONNX spec PR #5741) decrements the output extent when `ceil_mode` would produce an extra stride that never touches valid input. Due to this, the spec (in ONNX) for AveragePool and MaxPool were updated, but LpPool wasn't.
- The spec and reference tests now need to reflect that behavior.  
- The new tests ensure both shape inference and backend paths adhere to the standard set by Pytorch, so future regressions are caught.  
- **Fixes:** [#25848](https://github.com/microsoft/onnxruntime/issues/25848)

**Note:** PyTorch docs are also incorrect for LpPool in that they do not mention that *“Sliding windows that would start in the right padded region are ignored,”* despite the PyTorch implementation for LpPool adhering to that behavior. Here is the corresponding PyTorch PR I opened to update LpPool’s documentation so that its spec explicitly matches this PR’s update: https://github.com/pytorch/pytorch/pull/163186

### CODE REVIEW

File: onnx/backend/test/case/node/lppool.py [hunk 1]
```diff
@@ -256,6 +256,64 @@ def export_lppool_2d_strides() -> None:
...
+            constant_values=0,
+        )
+        y = pool(
+            padded,
+            x_shape,
+            kernel_shape,
+            strides,
+            out_shape,
+            "LPPOOL",
+            pads_required=pads_required,
```
Comment (gramalingam) [inline onnx/backend/test/case/node/lppool.py:305]: This looks a bit strange ... why does `pool` take a padded input as well as the pads attributes?

Comment (gramalingam) [reply onnx/backend/test/case/node/lppool.py:305]: Also: may be a shorter example with the output explicitly shown would help clarify the behavior to the reader. As it is, it is unclear to a reader what the output should be.

Comment (jonahbernard) [reply onnx/backend/test/case/node/lppool.py:305]: > *This looks a bit strange ... why does pool take a padded input as well as the pads attributes?*

The padded input is needed to demonstrate the effect of `ceil_mode=1` when padding is present. The `pads` attribute is required because this block in `pool`:

```python
window = padded[shape[0], shape[1]]
window_vals = np.array(
    [
        window[i]
        for i in list(
            itertools.product(
                *[
                    [
                        pixel
                        for pixel in range(
                            strides[i] * shape[i + 2],
                            strides[i] * shape[i + 2]
                            + (1 + (kernel[i] - 1) * dilations[i]),
                            dilations[i],
                        )
                        if pixel < x_shape[i + 2] + pads[i] + pads[spatial_size + i]
                    ]
                    for i in range(spatial_size)
                ]
            )
        )
    ]
)
```
must know which indices are out of bounds. Without pads, it would incorrectly filter out padded indices that should count in the average.

**In conclusion:**

With padded input but no pads, the pads wouldn’t be counted in the denominator.

With unpadded input and pads, you’d hit out-of-bounds errors.

Comment (jonahbernard) [reply onnx/backend/test/case/node/lppool.py:305]: > *Also: may be a shorter example with the output explicitly shown would help clarify the behavior to the reader. As it is, it is unclear to a reader what the output should be.*

Okay-- I simplified the tests to the simplest possible case.

**Comment by bparks-nv** [comment]
Two quick comments as the filer of the [onnxruntime issue](https://github.com/microsoft/onnxruntime/issues/25848) that kicked this off:

First, I'll mention that if the idea is to re-define the onnx spec to match onnxruntime (ORT) behavior there should be some consideration of the fact that it will cause currently-compliant implementations to need to adjust (e.g. TensorRT).

Second, if the decision is to re-define the spec, I think the documentation should provide updated formulas for calculating the output shapes and the shape checking code should also be updated so that `onnx.checker.check_model(model, full_check=True)` will match the changed spec. I don't see that in the current version of the PR

**Comment by jonahbernard** [comment]
@bparks-nv 


> First, I'll mention that if the idea is to re-define the onnx spec to match onnxruntime (ORT) behavior there should be some consideration of the fact that it will cause currently-compliant implementations to need to adjust (e.g. TensorRT).

This PR doesn't change the ONNX spec. The behavior that is _**explicitly**_ added to the ONNX docs in this PR was _**implicitly**_ added to the spec in Opset 22. Thus, no implementations that are compliant with Opset 22 should need to adjust. This PR simply brings it to light.

> Second, if the decision is to re-define the spec, I think the documentation should provide updated formulas for calculating the output shapes and the shape checking code should also be updated so that `onnx.checker.check_model(model, full_check=True)` will match the changed spec. I don't see that in the current version of the PR

`onnx.checker.check_model(model, full_check=True)` already implements the behavior that I am trying to include in the docs:

```   cc
 int64_t output_size =
        (effective_input_size - effective_kernel_shape[i] + (ceil_mode ? strides[i] - 1 : 0)) / strides[i] + 1;
    if (ceil_mode == 1 && (output_size - 1) * strides[i] >= (input_size + pads[i])) {
      // we need to match pytorch's behavior of "Sliding windows that would start in the right padded region are
      // ignored." (https://pytorch.org/docs/stable/generated/torch.nn.MaxPool1d.html#maxpool1d). this code follows the
      // same logic as PyTorch's C++ implementation:
      // https://github.com/pytorch/pytorch/blob/f1cdb39da3850c47d51ec6a5b1ae864c32b3accf/aten/src/ATen/native/Pool.h#L54C21-L54C21
      --output_size;
    }
```

Hence, this is why I mentioned above that this behavior is already baked into the ONNX spec, just implicitly. 

The reason `onnx.checker.check_model(model, full_check=True)` raised an exception in your original [onnxruntime issue](https://github.com/microsoft/onnxruntime/issues/25848) is that you had it set to opset 18, which does not have the "ignore last window if starting in padding" rule in ONNX (but opset 18 does have it in ONNX Runtime, which leads to deadlock in your case).

**Comment by bparks-nv** [comment]
>  The behavior that is explicitly added to the ONNX docs in this PR was implicitly added to the spec in Opset 22.

Got it, thanks for the clarification. In that case, my suggestion can be reduced to also providing a formula for when the size will be reduced by one, similar to what pytorch already provides in its [AvgPool2d](https://docs.pytorch.org/docs/stable/generated/torch.nn.AvgPool2d.html) documentation (text starting with "Per the note above, if ceil_mode is True ...").

**Comment by jonahbernard** [comment]
> Got it, thanks for the clarification. In that case, my suggestion can be reduced to also providing a formula for when the size will be reduced by one, similar to what pytorch already provides in its [AvgPool2d](https://docs.pytorch.org/docs/stable/generated/torch.nn.AvgPool2d.html) documentation (text starting with "Per the note above, if ceil_mode is True ...").

@bparks-nv done

**Comment by jonahbernard** [comment]
@gramalingam this is ready for your review, please. Thank you:)

**Comment by andife** [comment]
@jonahbernard could you fix dco?
